django/contrib/auth/mixins.py

```diff
@@ -0,0 +1,91 @@
+from django.conf import settings
+from django.contrib.auth import REDIRECT_FIELD_NAME
+from django.contrib.auth.views import redirect_to_login
+from django.core.exceptions import ImproperlyConfigured, PermissionDenied
+from django.utils import six
+from django.utils.encoding import force_text
+
+
+class UserPassesTestMixin(object):
+    """
+    Mixin for class-based views that checks that the user passes the given by
+    the class method `test_func`.
+    """
+    login_url = None
+    redirect_field_name = REDIRECT_FIELD_NAME
+    raise_exception = False
+
+    def test_func(self, user):
+        raise NotImplementedError(
+            '{0} is missing implementation of the '
+            'test_func method.'.format(self.__class__.__name__))
+
+    def get_test_func(self):
+        return getattr(self, "test_func")
+
+    def dispatch(self, request, *args, **kwargs):
+        user_test_result = self.get_test_func()(request.user)
+
+        if not user_test_result:
+            return self.handle_no_permission(request)
+
+        return super(UserPassesTestMixin, self).dispatch(
```

Comment:
You can join this and a couple of other lines. We have a max line length of 119 chars.